Exerscise_3/main.cpp

```diff
@@ -86,19 +87,54 @@ int main(int args, char **argv){
             std::cerr << ia.what() << std::endl;
             return EXIT_FAILURE;
         }
-        if (write_data_to_file("../Exam.txt", exam_vector)) {
+        if (args == 2) {
+            if (write_data_to_file("../Exam.txt", exam_vector)) {
+                return EXIT_FAILURE;
+            }
+            if (write_data_to_file("../Cource.txt", cource_vector)) {
+                return EXIT_FAILURE;
+            }
+            if (write_data_to_file("../Student.txt", student_vector)) {
+                return EXIT_FAILURE;
+            }
+            if (write_data_to_file("../Teacher.txt", teacher_vector)) {
+                return EXIT_FAILURE;
+            }
+            return EXIT_SUCCESS;
+        } else if (args > 2 and argv[2]==std::string("SHOW")) {
```

Comment:
I think that for the 4th tasks, it'd be good to introduce a new parser class, that parses the arguments and returns a specific implementation of `ICommand`. Its interface would look like this:
```cpp
class CommandParser
{
public:
    std::unique_ptr<ICommand> parse(std::vector<std::string> arguments) const;
};
```
So, this class would handle all the parsing and you would just call `parsed_command->execute();`